submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+                    <p>&nbsp;</p>
+                    <p><strong>User Location:</strong></p>
+                    <p>${user.location.country}</p>
+                    <p>${user.location.state}</p>
+                    <p>${user.location.city}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Contacts:</strong></p>
+                    <p>
+                        <a href="mailto:${user.email}">${user.email}</a>
+                    </p>
+                    <p>
+                        <a href="tel:${user.phone}">${user.phone}</a>
+                    </p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Gender:</strong></p>
+                    <p class="uppercase">${user.gender}</p>
+                </div>
+
+                <div class="user-button user-button--close">
+                    Back to Friends
+                </div>
+            </div>
+        </div>
+    `;
+    mainContainer.innerHTML = singleUser;
+}
+
+// filters
+let filter = document.querySelector('.filter__block input');
+nameInput.addEventListener('input', (e) => {
+    const value = e.preventDefault();
+    const pipes = [filterByName];
+    
+    if (getOpt('alphabet')) {
+        pipes.push(sortByAlphabet);
+    }
+    if (getOpt('age')) {
+        pipes.push(sortByAge);
+    }
+    if (getOpt('gender')) {
+        pipes.push(filterByGender);
+    }
+
+    renderUsers(pipes);
+});
+
+document.querySelectorAll('.filter__block input[type="radio"]').forEach((elem) => {
+    elem.addEventListener("change", function(event) {
+
+        switch (event.target.name) {
+            case 'alphabet':
+                uncheckRadio('age');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAlphabet
+                ]);
+
+                break;
+            case 'gender':
+                const pipes = [
+                    filterByName,
+                    filterByGender,
+                ];
+                if (getOpt('alphabet')) {
+                    pipes.push(sortByAlphabet);
+                }
+                if (getOpt('age')) {
+                    pipes.push(sortByAge);
+                }
+                renderUsers(pipes);
+
+                break;
+            case 'age':
+                uncheckRadio('alphabet');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAge
+                ]);
+                break;
+            default:
+                // renderUsers();
+        }
+    });
+});
+
+const filterByName = (user) => {
+    const searchValue = nameInput.value.trim().toLowerCase();
+    return user.name.first.toLowerCase().includes(searchValue);
+}
+
+const filterByGender = (user) => {
+    const genderRadioOpt = getOpt('gender');
+    if (!genderRadioOpt) { // Pass all if none gender checked
```

Comment:
You can just put a check into the `All` by default. Radiobuttons should always have a default value anyway.